### PR TITLE
refactor bot to be aware of when it pings

### PR DESCRIPTION
**Pull request by XAMPPRocky**

### CODE REVIEW

File: src/bot.rs [hunk 1]
```diff
@@ -200,308 +306,234 @@ impl Bot {
...
+
+    pub fn act(&mut self) -> Result<()> {
+        for (key, value) in self.db.iterator(IteratorMode::Start) {
+            let entry: DbEntry =
+                serde_json::from_str(&String::from_utf8_lossy(&value)).context(error::Json)?;
+            log::info!("Checking Entry: {:?}", entry);
+
+            match entry {
+                DbEntry::PullRequest(mut data) => {
+                    if data.n_participants >= 2 {
```
Comment (sjeohp) [inline src/bot.rs]: `n_participants` will currently never be set, right?

Comment (XAMPPRocky) [reply src/bot.rs]: It is currently set, see lines 462–472.

File: src/bot.rs [hunk 2]
```diff
@@ -200,308 +306,234 @@ impl Bot {
...
+            let prs = self.pull_requests(&repo)?;
+            for pr in prs {
+                let engineer = match self.engineers.get(&pr.user.login) {
+                    Some(eng) => eng,
+                    _ => continue,
+                };
+
+                let id_bytes = pr.id.to_be_bytes();
+                live_ids.push(id_bytes.clone());
+                let np = pr.requested_reviewers.len() as u32;
```
Comment (sjeohp) [inline src/bot.rs]: What would be the value of `pr.requested_reviewers.len()` if
- 3 reviewers were requested and all added comments/reviews
- no reviewers were requested but 3 people volunteered comments/reviews

Comment (XAMPPRocky) [reply src/bot.rs]: To answer the first would you be able to comment on this PR? https://github.com/sjeohp-test-org/test-repo/pull/9

Comment (sjeohp) [reply src/bot.rs]: Commented, although technically I already had since the bot was using my account

**Comment by XAMPPRocky** [comment]
I'm going to close this and reopen it after #17 has been merged.

**Pull request closed without merging**